### PR TITLE
Preserve correct TYPO3 comment and don't throw errors if no comment i…

### PR DESCRIPTION
**Pull request by theobscenezen**
…s set

### CODE REVIEW

File: Classes/Service/HtmlMinifier.php [hunk 1]
```diff
@@ -103,4 +103,22 @@ protected function isFeatureActive(string $feature): bool
...
+    private function preserveTypo3Comment(string $html): string
+    {
+        $typo3CommentStart = strpos($html, '<!--');
+        if ($typo3CommentStart !== false) {
+            $typo3CommentStop = strpos($html, '-->', $typo3CommentStart);
+            $typo3Comment = substr($html, $typo3CommentStart, $typo3CommentStop - $typo3CommentStart + 3);
+
+            if (strpos($typo3Comment, 'TYPO3') !== false) {
+                return $typo3Comment;
+            } else {
```
Comment (ChrisB9) [inline Classes/Service/HtmlMinifier.php]: the else here is not necessary thanks to the return statement

**Review by weakbit — APPROVED** [review]

File: Classes/Service/HtmlMinifier.php [hunk 2]
```diff
@@ -103,4 +103,22 @@ protected function isFeatureActive(string $feature): bool
...
+                $html = str_replace($typo3Comment, '', $html);
+                return $this->preserveTypo3Comment($html);
```
Comment (ChrisB9) [inline Classes/Service/HtmlMinifier.php]: doesn't that remove comments such as these: <!--[if lte IE 8]>

**Review by ChrisB9 — CHANGES_REQUESTED** [review]

File: Classes/Service/HtmlMinifier.php (hunk 2 above)
Comment (Kanti) [reply Classes/Service/HtmlMinifier.php]: or `<!--TYPO3SEARCH_begin-->` `<!--TYPO3SEARCH_end-->`

Comment (theobscenezen) [reply Classes/Service/HtmlMinifier.php]: i guess that's the sense of removing comments
but you are warmly invited to commit your ideas in a seperate bugfix branch

**Pull request merged**